Sources/Pathspec/GitIgnoreSpec.swift

```diff
@@ -8,14 +8,25 @@
 import Foundation
 
 struct GitIgnoreSpec: Spec {
+    enum Error: Swift.Error {
+        case emptyPattern
+        case containsPoundPrefix
+        case containsTrippleStar
+        case emptyRoot
+    }
+
 	private(set) var inclusive: Bool = true
+
+    let pattern: String
 	let regex: NSRegularExpression
-	
-	init?(pattern: String) {
-		guard !pattern.isEmpty else { return nil }
-		guard !pattern.hasPrefix("#") else { return nil }
-		guard !pattern.contains("***") else { return nil }
-		guard pattern != "/" else { return nil }
+
+    init(pattern: String) throws {
+        self.pattern = pattern
+
+        guard !pattern.isEmpty else { throw Error.emptyPattern }
+		guard !pattern.hasPrefix("#") else { throw Error.containsPoundPrefix }
```

Comment:
This one is good for now, but looking through the ruby version I think it might make sense that creating a spec instance from a commented out line wouldn't produce an error, but would instead produce a spec that would never be true/never be run. I'll do that in a subsequent pass though (unless you're keen on it)